src/virtual_dom/mod.rs

```diff
@@ -70,7 +72,8 @@ impl Classes {
     ///
     /// Takes the logical union of both `Classes`.
     pub fn extend<T: Into<Classes>>(mut self, other: T) -> Self {
-        self.set.extend(other.into().set.into_iter());
+        self.set
+            .extend(other.into().set.into_iter().filter(|c| !c.is_empty()));
```

Comment:
I feel like the filter empty check isn't needed here if a Classes instance already has stripped away all empty strings.